### ISSUE TITLE
memory leak

### ISSUE THREAD

**Issue by solos** [mulaRahul/keyviz#334]
**Describe the bug**
It used 12GB memory

**Version**: v2.0.0alpha2

**To Reproduce**

just use long time, not closed.

**Expected behavior**
used too much memory

**Screenshots**
![image](https://github.com/user-attachments/assets/c6f78a64-5f35-4854-bca4-891db92af899)

**Desktop (please complete the following information):**
 - macOS 13.6.7

**Additional context**

**Comment by nitwoo** [comment]
Yeah noticed the same, my performance was getting really bad after a while while recording stuff for my students.

**Comment by tudan110** [comment]
the same

**Comment by mulaRahul** [comment]
Has anybody experienced the same on Windows?
